test/nativetest/artifacts_posix_test.go

```diff
@@ -11,144 +11,144 @@ import (
+					"artifacts": [
+						{
+							"name": "public/result.txt",
+							"type": "file",
+							"path": "` + filename + `.txt"
+						}
+					]
+				}`,
+				Artifacts: workertest.ArtifactAssertions{
+					"public/logs/live.log":         workertest.ReferenceArtifact(),
+					"public/logs/live_backing.log": workertest.GrepArtifact("hello-world"),
+					"public/result.txt":            workertest.GrepArtifact("42"),
+				},
+			}, {
+				Title:   "Artifact Directory",
+				Success: true,
+				Payload: `{
+					"command": ["sh", "-c", "` + strings.Join([]string{
+					"echo 'hello-world'",
+					"mkdir -p sub/subsub/subsubsub",
+					"echo 42 > sub/subsub/result.txt",
+					"echo -n '<html></html>' > sub/subsub/subsubsub/result.html",
+				}, " && ") + `"],
+					"env": {},
+					"maxRunTime": "10 minutes",
+					"artifacts": [
+						{
+							"name": "public",
+							"type": "directory",
+							"path": "sub"
+						}
+					]
+				}`,
+				Artifacts: workertest.ArtifactAssertions{
+					"public/logs/live.log":                workertest.ReferenceArtifact(),
+					"public/logs/live_backing.log":        workertest.GrepArtifact("hello-world"),
+					"public/subsub/result.txt":            workertest.GrepArtifact("42"),
+					"public/subsub/subsubsub/result.html": workertest.MatchArtifact("<html></html>", "text/html; charset=utf-8"),
+				},
+			}, {
+				Title:   "Artifact Directory Is File",
+				Success: false,
+				Payload: `{
+					"command": ["sh", "-c", "echo 42 > notafolder"],
+					"env": {},
+					"maxRunTime": "10 minutes",
+					"artifacts": [
+						{
+							"name": "public/myfolder",
+							"type": "directory",
+							"path": "notafolder"
+						}
+					]
+				}`,
+				Artifacts: workertest.ArtifactAssertions{
+					"public/logs/live.log":         workertest.ReferenceArtifact(),
+					"public/logs/live_backing.log": workertest.GrepArtifact("notafolder"),
+					"public/myfolder":              workertest.ErrorArtifact(),
+				},
+			}, {
+				Title:   "Artifact File Is Directory",
+				Success: false,
+				Payload: `{
+					"command": ["sh", "-c", "mkdir -p sub/subsub/ && echo 42 > sub/subsub/result.txt"],
+					"env": {},
+					"maxRunTime": "10 minutes",
+					"artifacts": [
+						{
+							"name": "public/subsub",
+							"type": "file",
+							"path": "sub/subsub"
+						}
+					]
+				}`,
+				Artifacts: workertest.ArtifactAssertions{
+					// Expect some error message saying "sub/subsub"
+					"public/logs/live.log":         workertest.ReferenceArtifact(),
+					"public/logs/live_backing.log": workertest.GrepArtifact("sub/subsub"),
+					"public/subsub":                workertest.ErrorArtifact(),
+				},
+			}, {
+				Title:   "Artifact File Not Found",
+				Success: false,
+				Payload: `{
+					"command": ["true"],
+					"env": {},
+					"maxRunTime": "10 minutes",
+					"artifacts": [
+						{
+							"name": "public/result.txt",
+							"type": "file",
+							"path": "no-such-file-` + filename + `.txt"
```

Comment:
Where is the variable filename defined?

Reply:
Probably higher up...this is just a mechanically refactored test case.. see the commit